types/nearley/index.d.ts

```diff
@@ -47,26 +49,46 @@ export interface ParserOptions {
 }
 
 export class Rule {
-    constructor(name: any, symbols: any, postprocess: any);
-    toString(withCursorAt: any): any;
     static highestId: number;
+
+    id: number;
+    name: string;
+    symbols: any[];
+    postprocess?(data: any[], reference: number, wantedBy: any): void;
+
+    constructor(name: string, symbols: any[], postprocess?: (data: any[], reference: number, wantedBy: any) => void);
+
+    toString(withCursorAt?: number): string;
 }
 
 export class Grammar {
-    constructor(rules: Rule[], start: string);
+    static fromCompiled(rules: CompiledRules, start?: string): Grammar;
```

Comment:
Prefer `fromCompiled(rules: CompiledRules)`: the `start` option is deprecated, and retained only for backwards compat.

Reply:
Didn't know about that, it's not mentioned in the source. Will remove.